lib/neo4j.rb

```diff
@@ -86,6 +86,13 @@
 require 'neo4j/active_node/scope'
 require 'neo4j/active_node'
 
+require 'neo4j/migration'
+require 'neo4j/migrations/helpers'
+require 'neo4j/migrations/migration_file'
+require 'neo4j/migrations/base'
+require 'neo4j/migrations/runner'
+require 'neo4j/migrations/schema_migration'
+
```

Comment:
I feel like this `require` practice is growing a bit too much ;)  I wonder about using Ruby's `autoload`:

http://www.rubyinside.com/ruby-techniques-revealed-autoload-1652.html

Migrations will generally run in the rake task process and then the process dies.  So if we use `autoload` we (I think) won't get the `Migration` code loaded into memory in the application, just in the rake task.

If it works well here, I think it's something that we see if it makes sense elsewhere.